### DISCUSSION TITLE
Correlation ID

### DISCUSSION THREAD

**Discussion by johncmunson** [prisma/prisma#15259] [Q&A] [answered]
## Question
Does Prisma support the concept of a correlation id? If not, might this be possible to implement in userland via middleware or a plugin?

The concept is simple... when you are trying to debug slow queries in production (e.g. RDS or similar), it can sometimes be difficult to know exactly where in your source code that the query is coming from. A correlation id would make debugging much simpler.

For example, this Prisma query...
```ts
const user = await prisma.user.findUnique({
  where: { id: 2 },
  correlationId: 't5kdb8a',
})
```
Could generate this SQL...
```sql
/* id: t5kdb8a */ SELECT * FROM user WHERE id = 2;
```

Seeing the correlation id show up in tools like the RDS query analyzer would be super beneficial.

![Screen Shot 2022-09-08 at 2 34 50 PM](https://user-images.githubusercontent.com/19480078/189390021-8a1ed42a-c0c9-47e0-b093-176f51317065.png)

## Prior Art
Google has a very similar project called [Sqlcommenter](https://cloud.google.com/blog/topics/developers-practitioners/introducing-sqlcommenter-open-source-orm-auto-instrumentation-library) that integrates with various ORMs such as [Knex](https://github.com/google/sqlcommenter/blob/master/nodejs/sqlcommenter-nodejs/packages/sqlcommenter-knex/README.md). Unfortunately, there does not exist an adapter for Prisma.

**Comment by johncmunson** [comment, marked as answer]
Mkay, I found an older issue [here](https://github.com/prisma/prisma/issues/10746) relating to the same topic.

**Comment by austincrim** [comment]
Hey @johncmunson 👋,

If you're interested in seeing this feature added to Prisma, definitely add a comment on that linked issue with more details about your use case.

You might also want to take a look at the [Prisma Tracing feature ](https://www.prisma.io/docs/concepts/components/prisma-client/opentelemetry-tracing) which you might find valuable.
